app/_src/networking/meshexternalservice.md

```diff
@@ -0,0 +1,352 @@
+Protocols that are supported are: `tcp`, `grpc`, `http`, `http2`.
+
+```yaml
+match:
+  type: HostnameGenerator
+  port: 4244
+  protocol: tcp
+```
+
+### Endpoints
+
+This section specifies the destination of the matched traffic.
+It's possible to define IPs, DNS names and unix domain sockets.
+
+```yaml
+endpoints:
+  - address: 1.1.1.1
+    port: 12345
+  - address: example.com
+    port: 80
+  - address: unix:///tmp/example.sock
+```
+
+### TLS
+
+This section describes the TLS and verification behaviour.
+You can define TLS version requirements, option to allow renegotiation, verification of SNI, SAN, custom CA and client certificate and key for server verification.
+To disable parts of the verification you can set different `mode` - `SkipSAN`, `SkipCA`, `SkipAll`, `Secured` (default).
+
+```yaml
+tls:
+  version:
+    min: TLS12
+    max: TLS13
+  allowRenegotiation: false
+  verification:
+    mode: SkipCA
+    serverName: "example.com"
+    subjectAltNames:
+      - type: Exact
+        value: example.com
+      - type: Prefix
+        value: "spiffe://example.local/ns/local"
+    caCert:
+      inline: dGVzdA==
+    clientCert:
+      secret: "123"
+    clientKey:
+      secret: "456"
+```
+
+When TLS is enabled but `caCert` is not set, the sidecar uses the [autodetected OS-specific CA](https://github.com/kumahq/kuma/blob/aba6518fca65bc7ab52e5328eb686a51a6f98a53/app/kuma-dp/pkg/dataplane/certificate/cert.go#L12).
+The user can override the default CA by setting the path in the environment variable `KUMA_DATAPLANE_RUNTIME_DYNAMIC_SYSTEM_CA_PATH` for the sidecar.
+
+### DNS setup
+
+To be able to access `MeshExternalService` via a hostname you need to define a [HostnameGenerator](/docs/{{ page.version }}/policies/hostnamegenerator) with a `meshExternalService` selector.
+
+{% warning %}
+Do **not** hijack original addresses like httpbin.com (the way it was done with [External Service](/docs/{{ page.version }}/policies/external-services)).
+If you need to transparently pass traffic through the Mesh without modifying it use [MeshPassthrough](/docs/{{ page.version }}/policies/meshpassthrough).
+{% endwarning %}
+
+For clients that the hostname is not fully known in advance please take a look at [Wildcard DNS matching in MeshPassthrough](/docs/{{ page.version }}/policies/meshpassthrough/#wildcard-dns-matching).
+
+## Examples
+
+TCP examples use https://tcpbin.com/ service which is a TCP echo service, check out the website for more details.
+HTTP examples use https://httpbin.org/ service which is a website for inspecting and debugging HTTP requests.
+GRPC examples use https://grpcbin.test.k6.io/ service which is a gRPC Request & Response Service.
+You can use [grpcurl](https://github.com/fullstorydev/grpcurl) as a client, it is available in [netshoot](https://github.com/nicolaka/netshoot) debug image alongside other tools used in later sections.
+
+For the examples below the following `HostnameGenerator` will be used:
```

Comment:
So there's no default hostname generator atm?

Reply:
I think te answer is here https://github.com/kumahq/kuma/pull/10478

Reply:
Might be worth stating this in the docs that in future versions of kuma there will be a default HostnameGenerator

Reply:
And write the docs in a way that will be easy to make this part optional

Reply:
> And write the docs in a way that will be easy to make this part optional

So you'd use the template that is suggested currently so we don't have to if/else the examples? Otherwise I'd just if/else https://github.com/kumahq/kuma-website/pull/1816/files#diff-8d43902ad147c9062f03d2eea50710e9ba83961b78556a27086f0e8b0b8a0d05R98-R131 this part.